eventhubs/client/azure-eventhubs/src/main/java/com/azure/eventhubs/EventData.java

```diff
@@ -77,6 +86,54 @@ public EventData(ByteBuffer data) {
         }
 
         this.data = data;
+        this.properties = new HashMap<>();
+    }
+
+    /*
+     * Creates an event from a message
+     */
+    EventData(Message message) {
```

Comment:
nit:  should this an an explicit visibility modifier?   Seems that you're using them on elsewhere.

Reply:
In .NET, they have the "internal" modifier. In Java, they have this concept of "package-private", where no modifier specified means "package-private". (ie. only classes in the same package namespace can see it.) It's useful for testing, so we don't have to do that "InternalsVisibleTo" when testing
https://docs.oracle.com/javase/tutorial/java/javaOO/accesscontrol.html